examples/rust/domains/discovery.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use clap::Parser;
+use iceoryx2::prelude::*;
+use iceoryx2_bb_log::{set_log_level, LogLevel};
+use iceoryx2_bb_system_types::file_name::*;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let args = parse_args();
+
+    // create a new config based on the global config
+    let mut config = Config::global_config().clone();
+
+    // The domain name becomes the prefix for all resources.
+    // Therefore, different domain names never share the same resources.
+    config.global.prefix = FileName::new(args.domain.as_bytes())?;
```

Comment:
So this line is the only significant change that users need to make to leverage this feature. Nice and simple.